test/integration/M3/Base.t.sol

```diff
@@ -0,0 +1,219 @@
+            vm.startPrank(ProxyAdmin(getTransparentUpgradeableProxyAdminAddress(address(yneth))).owner());
+            ProxyAdmin(
+                getTransparentUpgradeableProxyAdminAddress(address(yneth))
+            ).upgradeAndCall(
+                ITransparentUpgradeableProxy(address(yneth)),
+                address(new ynETH()),
+                ""
+            );
+            vm.stopPrank();
+        }
+
+        // deploy ynETHRedemptionAssetsVault
+        {
+            TransparentUpgradeableProxy _proxy = new TransparentUpgradeableProxy(
+                address(new ynETHRedemptionAssetsVault()),
+                GLOBAL_ADMIN,
+                ""
+            );
+            ynETHRedemptionAssetsVaultInstance = ynETHRedemptionAssetsVault(payable(address(_proxy)));
+        }
+
+        // deploy WithdrawalQueueManager
+        {
+            TransparentUpgradeableProxy _proxy = new TransparentUpgradeableProxy(
+                address(new WithdrawalQueueManager()),
+                GLOBAL_ADMIN,
+                ""
+            );
+            ynETHWithdrawalQueueManager = WithdrawalQueueManager(address(_proxy));
+        }
+
+        // initialize ynETHRedemptionAssetsVault
+        {
+            ynETHRedemptionAssetsVault.Init memory _init = ynETHRedemptionAssetsVault.Init({
+                admin: GLOBAL_ADMIN,
+                redeemer: address(ynETHWithdrawalQueueManager),
+                ynETH: IynETH(address(yneth))
+            });
+            ynETHRedemptionAssetsVaultInstance.initialize(_init);
+        }
+
+        // initialize WithdrawalQueueManager
+        {
+            WithdrawalQueueManager.Init memory managerInit = WithdrawalQueueManager.Init({
+                name: "ynETH Withdrawal Manager",
+                symbol: "ynETHWM",
+                redeemableAsset: IRedeemableAsset(address(yneth)),
+                redemptionAssetsVault: IRedemptionAssetsVault(address(ynETHRedemptionAssetsVaultInstance)),
+                admin: GLOBAL_ADMIN,
+                withdrawalQueueAdmin: GLOBAL_ADMIN,
+                redemptionAssetWithdrawer: GLOBAL_ADMIN,
+                requestFinalizer:  GLOBAL_ADMIN,
+                withdrawalFee: 500, // 0.05%
+                feeReceiver: GLOBAL_ADMIN
+            });
+            ynETHWithdrawalQueueManager.initialize(managerInit);
+        }
+
+        // initialize stakingNodesManager withdrawal contracts
+        {
+            StakingNodesManager.Init2 memory initParams = StakingNodesManager.Init2({
+                redemptionAssetsVault: ynETHRedemptionAssetsVaultInstance,
+                withdrawalManager: GLOBAL_ADMIN,
+                stakingNodesWithdrawer: GLOBAL_ADMIN
+            });
+            
+            vm.prank(GLOBAL_ADMIN);
+            stakingNodesManager.initializeV2(initParams);
+        }
+
+        // upgrade StakingNodeImplementation
+        {
+            stakingNodeImplementation = new StakingNode();
+            vm.prank(GLOBAL_ADMIN);
```

Comment:
This should prank STAKING_ADMIN

Reply:
https://github.com/yieldnest/yieldnest-protocol/pull/159/commits/48f72390ec79e2ad439f89d86f244eb6ebc0c7e5